app/javascript/packs/unpublishPostModal.jsx

```diff
@@ -0,0 +1,149 @@
+    });
+
+    const outcome = await response.json();
+
+    /* eslint-disable no-restricted-globals */
+    if (outcome.message == 'success') {
+      window.top.location.assign(`${window.location.origin}${outcome.path}`);
+    } else {
+      top.addSnackbarItem({
+        message: `Error: ${outcome.message}`,
+        addCloseButton: true,
+      });
+    }
+  } catch (error) {
+    top.addSnackbarItem({
+      message: `Error: ${error}`,
+      addCloseButton: true,
+    });
+  }
+
+  toggleUnpublishPostModal();
+}
+
+/**
+ * Shows or hides the flag user modal.
+ */
+export function toggleUnpublishPostModal() {
+  const modalContainer = top.document.getElementsByClassName(
+    'unpublish-post-modal-container',
+  )[0];
+  modalContainer.classList.toggle('hidden');
+
+  if (!modalContainer.classList.contains('hidden')) {
+    top.window.scrollTo(0, 0);
+    top.document.body.style.height = '100vh';
+    top.document.body.style.overflowY = 'hidden';
+  } else {
+    top.document.body.style.height = 'inherit';
+    top.document.body.style.overflowY = 'inherit';
+  }
+}
+
+/**
+ * Initializes the Unpublish Post modal for the given article ID, author username and article slug.
+ *
+ * @param {number} articleId
+ * @param {string} authorUsername
+ * @param {string} articleSlug
+ */
+export function initializeUnpublishPostModal(
+  articleId,
+  authorUsername,
+  articleSlug,
+) {
+  // Check whether context is ModCenter or Friday-Night-Mode
+  const modContainer = document.getElementById('mod-container');
+
+  if (!modContainer) {
+    return;
+  }
+
+  render(
+    <UnpublishPostModal
+      articleId={articleId}
+      authorUsername={authorUsername}
+      articleSlug={articleSlug}
+    />,
+    document.getElementsByClassName('unpublish-post-modal-container')[0],
+  );
+}
+
+/**
+ * A modal for unpublishing a post. This can be used in the moderation center
+ * or on an article page.
+ *
+ * @param {number} props.articleId ID of the article to be unpublished.
+ * @param {string} props.authorUsername Username of the article's author.
+ * @param {string} props.articleSlug Slug of the article to be unpublished.
+ */
+export function UnpublishPostModal({ articleId, authorUsername, articleSlug }) {
+  return (
+    <div
+      data-testid="unpublish-post-modal"
+      class="crayons-modal crayons-modal--small absolute unpublish-post-modal"
+    >
+      <div class="crayons-modal__box">
+        <header class="crayons-modal__box__header unpublish-post-modal-header">
+          <h2 class="crayons-modal__box__header__title">Unpublish post</h2>
+          <Button
+            icon={RemoveIcon}
+            className="inline-flex"
+            onClick={toggleUnpublishPostModal}
+          />
+        </header>
+        <div class="crayons-modal__box__body">
+          <div class="grid gap-4">
+            <p>
+              Once unpublished, this post will become invisible to the public
+              and only accessible to Anuj.
```

Comment:
Is this a placeholder for now? Since the feature is behind a feature flag I don't see a problem with this, but we could interpolate `authorUsername` here unless I'm missing some context 🙂

Reply:
Howwwwwww did I miss this???? 🤦🏾‍♀️  Thank you so much!